funcbench/env.go

```diff
@@ -89,10 +103,24 @@ type GitHub struct {
 }
 
 func newGitHubEnv(ctx context.Context, e environment, gc *gitHubClient, workspace string) (Environment, error) {
-	r, err := git.PlainCloneContext(ctx, fmt.Sprintf("%s/%s", workspace, gc.repo), false, &git.CloneOptions{
-		URL:      fmt.Sprintf("https://github.com/%s/%s.git", gc.owner, gc.repo),
-		Progress: os.Stdout,
-	})
+
+	var r *git.Repository
+	var err error
+	// Retry 10 times.
+	for i := 1; i <= 10; i++ {
+		if err := os.RemoveAll(filepath.Join(workspace, gc.repo)); err != nil {
+			return nil, err
+		}
+		e.logger.Println("Cloning ", gc.owner, ":", gc.repo, " is in progress. Checking in ", provider.GlobalRetryTime)
+		time.Sleep(provider.GlobalRetryTime)
```

Comment:
This is a different Retry and shouldn't be mixed with when applying resources.

provider.GlobalRetryTime - this is how long to wait for retrying to apply a resource
and the retry here is how long to wait for cloning a repo. 

I would say lets keep these separate and have this hard coded or a separate env just for this package.

Also always think about dependencies. There is absolutely no reason why would the funcbench package to depend on the provider package.